### ISSUE TITLE
Colors in Cyanine Theme

### ISSUE THREAD

**Issue by josephzitt** [typemill/typemill#204]
(My apologies if this isn't the right place to ask. I'm relatively new to Github, which seems to be more for developer questions than user questions, but I don't know of a more appropriate forum.)

I'm trying to figure out the color scheme in the default Cyanine theme. Changing the values under Colors in Setup is having unexpected results. I don't quite follow the descriptions. I would like to do a sort of Dark Mode design, with pale grey text on a dark grey background, which I find easiest to read. As an example, http://darkforcesswing.blogspot.com/ has a relatively good example of such a color scheme. I understand how the notation for colors work, but I don't quite understand which values to set.

What would I set for these?

I'm primarily a tech writer rather than a coder, so I would be happy to help with rewording the descriptions once I understand how it works.

**Comment by josephzitt** [comment]
OK, I have the colors as I want them with some further tweaking.

But now the contents have gotten messed up. Under content/02-book-3-in-the-days-of-the-virus, there are folders for each month. When I try to navigate through them or look at them in the sitemap, March appears twice, once in the proper place and once after August. Only the second instance has content. The rest of the months don't appear to have content either, although the content is there. This may be a reslut of having changed things from pages to posts and back, but I'm confused as to how to fix it.

On another issue (should I open another thread for this?): I see that there is code in cyanine/page.twig to handle widgets, right where I would expect to find it. As I'd mentioned, I would like to drop in a block of markdown with a static blurb and links. It appears that I could do that by creating a "widgets" array somewhere with a call to a Markdown to HTML convertor and the content. It looks like several of these exist. Does Typemill include or support any of them? And where would I create the arry? It wold seem that it would be cleanest to put it in another file so as not to mess with page.twig itself, but (being new to much of this), I'm not clear where, Is this possible?

**Comment by josephzitt** [comment]
Hmm. An edit from yesterday on my previous comment appears to have disappeared.

I deleted everything from the scrambled Book 3 section and uploaded it again. That appears to have cleared up the out-of-order posts. 

I have also changed the colors to correspond to what Gmail's dark mode uses. Everything looks pretty good, except for the text on the expand-or-contract button (currently black on dark grey, so unreadable; I'm not sure what controls that) and that the body text, which looks good on my PC and my Android phone, is dim and blurry on my Mac.

**Comment by trendschau** [comment]
good to hear! Generally you can inspect the elements with the developer tool of your browser and then find out which css rule you need to change the design. For the button you can use this for example: 

````
.contentnav button {
    color: #fff;
}
````

Regarding the widget: That feature is not very sophisticated yet and you have to code a little plugin to use it. I suppose that is out of the scope. However, you can find the minimal documentation here:  https://typemill.net/theme-developers/theme-functions/render-widgets and you can check the search plugin that uses it: https://plugins.typemill.net/search

The part that uses the widget is this one: 

````
	public function onPageReady($page)
	{
		$pageData = $page->getData($page);

		$pageData['widgets']['search'] = '<div class="searchContainer" id="searchForm">'.
	        									'<input id="searchField" type="text" placeholder="search ..." />'.
	        									'<button id="searchButton" type="button">GO</button>'.
    									'</div>';
 		$page->setData($pageData);
	}
````

... but you have to adopt the html to your need and embed it into a whole working plugin. Instead of html you can also create a markdown field for your plugin, use that content, transform it into html and then inject it in the way above. This way you can edit your widget-content in the plugin-settings easily.

An alternative way is to add a new field to the theme and use it directly with the twig-markdown function (e.g. `markdown(myfieldcontent)` ), but that solution is not ideal because then you cannot update your theme without loosing that function or copy it manually again and again. 

Finally you have to style your widget-content (if it does not look good out of the box). You can either inject a CSS file with the plugin or you can use the custom-css-field of the cyanine-theme again. 

So it is a bit of work, easy for a developer but it requires developer skills. Typemill is not a page-builder and of course not so sofisticated like WordPress or other big CMS. I think it is very flexible already but the possibilities are limited for non-devs.

**Comment by josephzitt** [comment]
OK, flexing my minimal dev skills after reading the documentation...

I've created a plugin named "blurbbox" by making a copy of "search" and stripping it down to what's needed: Just the .yaml and .php files in a plugin directory of that name.

biurbbox.php is
```php
<?php

namespace Plugins\blurbbox;

use \Typemill\Plugin;
use \Typemill\Models\Write;

class blurbbox extends plugin
{
	protected $item;
	
    public static function getSubscribedEvents()
    {
		return array(
			'onPageReady'			=> 'onPageReady',
		);
	}
	
	public function onPageReady($page)
	{
		$pageData = $page->getData($page);

		$pageData['widgets']['blurbbox'] = '<div><p>Notes on (mostly) daily perceptions and experiences by Joseph Zitt, an American immigrant to Israel.</p>' .
											'<p><a href="http://www.josephzitt.com/home/books/as-if-in-dreams-notes-following-aliyah/">Buy the book.</a></p>' .
											'<p><a href="https://buttondown.email/josephzitt">Subscribe to the weekly newsletter.</a></p></div>'; 

 		$page->setData($pageData);
	}
}
```
blurbbox.yaml is
```yaml
name: blurbbox
version: 1.0.0
description: Add a blurb to site pages. Adapted from Search by Trendschau.
author: Joseph Zitt
homepage: https://www.josephzitt.com
licence: MIT
paypal: https://paypal.me/josephzitt
amount: 10
```
For a while, until I fixed some of the HTML embedded in the .php file, it prevented pages from loading at all (so at least I know that it was being noticed). Now that I've fixed that, nothing happens at all. It also doesn't show up in the Settings plugins page so that I could activate it.

I suspect that I've missed something simple. Do you spot anything?

**Comment by trendschau** [comment]
Hi Joseph, 

congratulations, it looks good! I copied your code and it works fine on my local version. Can you check if your folders and files are named correctly?

````
\blurbbox
  blurbbox.php
  blurbbox.yaml
````
You can also skip out some stuff from your code, but that is not the reason for the error: 

````
<?php

namespace Plugins\blurbbox;

use \Typemill\Plugin;

class blurbbox extends plugin
{	
    public static function getSubscribedEvents()
    {
		return array(
			'onPageReady'			=> 'onPageReady',
		);
	}
	
	public function onPageReady($page)
	{
		$pageData = $page->getData($page);

		$pageData['widgets']['blurbbox'] = '<div><p>Notes on (mostly) daily perceptions and experiences by Joseph Zitt, an American immigrant to Israel.</p>' .
											'<p><a href="http://www.josephzitt.com/home/books/as-if-in-dreams-notes-following-aliyah/">Buy the book.</a></p>' .
											'<p><a href="https://buttondown.email/josephzitt">Subscribe to the weekly newsletter.</a></p></div>'; 

 		$page->setData($pageData);
	}
}

````

If you still get an error, can you switch on the error reporting by adding this line: 

````
displayErrorDetails: true 
````
into the settings file `\settings\settings.yaml` and then post the error? 

If it works, you can style the box with the id #blurbbox, so something like 

````
#blurbbox p{ font-size: 1.2em} 
````


I will be off until Tuesday and then check again.

**Comment by josephzitt** [comment]
It works! The problem was that I had somehow nested a second blurbbox folder within the first one, with the files inside of that. (I probably fumbled something with FileZilla.) I have moved them up into the correct folder, and it's working. I'll have to fiddle with colors, but that's relatively easy.

Thanks for your help and encouragement.

**Comment by trendschau** [comment]
that is great and congratulations! I will be unavailable for some days but if you want you can enhance your plugin a bit and make the content editable with markdown. It is described in the tutorial here: https://typemill.net/plugin-developers/tutorial/make-variables-editable

Not important of course, but might be more convenient if you want to change the text in future :)

**Comment by trendschau** [comment]
Version 1.4.0 of typemill is published and also a version 1.1.0 of the cyanine theme. Color-Logic has been refactored completely, so you will have to update your individual color scheme. It should be possible to create that individual theme design without any additional custom css.
